src/anyio/_core/_contextmanagers.py

```diff
@@ -4,12 +4,20 @@
 from contextlib import AbstractAsyncContextManager, AbstractContextManager
 from inspect import isasyncgen, iscoroutine, isgenerator
 from types import TracebackType
-from typing import Generic, TypeVar, final
+from typing import Protocol, TypeVar, final
 
 _T_co = TypeVar("_T_co", covariant=True)
 
 
-class ContextManagerMixin(Generic[_T_co]):
+class _SupportsCtxMgr(Protocol[_T_co]):
+    def __contextmanager__(self) -> AbstractContextManager[_T_co]: ...
+
+
+class _SupportsAsyncCtxMgr(Protocol[_T_co]):
+    def __asynccontextmanager__(self) -> AbstractAsyncContextManager[_T_co]: ...
+
+
+class ContextManagerMixin:
```

Comment:
Removing the typevar from the class level here is not necessary but avoids having to declare another one to use in the method and since the mixin class doesn't really depend on the value type it's not needed.